ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPublic.qll

```diff
@@ -359,6 +385,11 @@ private module Cached {
     )
   }
 
+  cached
+  predicate methodHasSuperCall(MethodNode method, CallNode call) {
```

Comment:
Right now there aren't any other uses than in API graphs, so it doesn't matter. But in principle I'd say yes, it's a good place to cache, because the predicate is very small, and if re-evaluated at the wrong time it can trigger re-evaluation of `getEnclosingMethod` which is large and not cached.